chatkit-core/src/main/kotlin/com/pusher/chatkit/ChatkitCore.kt

```diff
@@ -0,0 +1,77 @@
+package com.pusher.chatkit
+
+import com.pusher.chatkit.api.createPlatformClientModule
+import com.pusher.chatkit.state.createStoreModule
+import com.pusher.chatkit.users.api.createUserSubscriberModule
+import com.pusher.util.Result
+import com.pusher.util.asSuccess
+import elements.Error
+import org.koin.core.KoinApplication
+import org.koin.core.module.Module
+import org.koin.dsl.koinApplication
+import org.koin.dsl.module
+
+/**
+ * The entry point for JVM apps to connect with the backend service
+ * and retrieve [ChatkitCore] instance for further interaction with the SDK.
+ */
+class ChatkitCoreConnector {
+
+    /**
+     * Creates instance of this class for your Chatkit instance
+     * using passed [TokenProvider][com.pusher.platform.tokenProvider.TokenProvider]
+     * (for authentication and user identification) and additional optional dependencies.
+     *
+     * @param instanceLocator available in the [dashboard](https://dash.pusher.com/chatkit/)
+     * (under the Credentials tab of a selected instance)
+     * @param dependencies used for passing your [TokenProvider][com.pusher.platform.tokenProvider.TokenProvider]
+     * alongside optional dependencies like your [Logger][com.pusher.platform.logger.Logger]
+     */
+    constructor(
+        instanceLocator: String,
+        dependencies: ChatkitDependencies
+    ) : this(instanceLocator, dependencies, overrideModules = emptyList())
+
+    internal constructor(
+        instanceLocator: String,
+        dependencies: ChatkitDependencies,
+        overrideModules: List<Module>
+    ) {
+        this.instanceLocator = instanceLocator
+        this.dependencies = dependencies
+
+        chatkitKoinApplication = koinApplication {
+            modules(
+                createPlatformClientModule(instanceLocator, dependencies),
+                module { single { dependencies.logger } },
+                createUserSubscriberModule(),
+
+                createStoreModule()
+            )
+            modules(overrideModules)
+        }
+    }
+
+    private val instanceLocator: String
+    private val dependencies: ChatkitDependencies
+    private val chatkitKoinApplication: KoinApplication
+
+    /**
+     * Asynchronously connects with the backend and provide [ChatkitCore] as the entry point
+     * to interact with the SDK.
+     *
+     * @param resultHandler callback for handling successful connection, or errors while
+     * trying to connect
+     */
+    fun connect(resultHandler: (Result<ChatkitCore, Error>) -> Unit) {
+        // ...
+        resultHandler(ChatkitCore().asSuccess())
+    }
+}
+
+/**
+ * The entry point to interact with the SDK after connecting.
+ *
+ * Use [ChatkitCoreConnector] to retrieve an instance of this class.
```

Comment:
Instead of saying `Use [ChatkitCoreConnector] to retrieve an instance of this class.` - how about linking to the method that contains that info e.g. Use [ChatkitCoreConnector method] to retrieve an instance of this class?

Reply:
ChatkitCoreConnector has to be constructed first and it does have only one public method so I think it's fine like that.